src/movies.js

```diff
@@ -1,16 +1,51 @@
 // Iteration 1: All directors? - Get the array of all directors.
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors.
+
+const movies = require('./data');
+
 // How could you "clean" a bit this array and make it unified (without duplicates)?
-function getAllDirectors() {}
+function getAllDirectors(movies) {
+  const allDirectors = movies.map(function (film) {
+    return film.director;
+  });
+  return allDirectors;
+}
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
-function howManyMovies() {}
+function howManyMovies(movies) {
+  const dramaSpielberg = movies.filter(function (film) {
+    return film.director === 'Steven Spielberg' && film.genre.includes('Drama');
+  });
+  if (dramaSpielberg.length === 0) {
+    return 0;
+  } else {
+    return dramaSpielberg.length;
+  }
+}
 
 // Iteration 3: All scores average - Get the average of all scores with 2 decimals
-function scoresAverage() {}
+function scoresAverage(movies) {
+  if (movies.length === 0) {
+    return 0;
+  } else {
+    scoresArray = movies.map(function (film) {
+      return film.score;
+    });
+  }
+  const totalScores = scoresArray.reduce(function (acc, curr) {
+    return acc + curr;
+  }, 0);
+  const averageScore = Math.round((totalScores / movies.length) * 100) / 100;
+  return averageScore;
+}
 
 // Iteration 4: Drama movies - Get the average of Drama Movies
-function dramaMoviesScore() {}
+function dramaMoviesScore(movies) {
+  const dramaMovies = movies.filter(function (film) {
+    return film.genre.includes('Drama');
+  });
+  return scoresAverage(dramaMovies);
+}
```

Comment:
The task is also asking to check if your movie object has score property. So first you need to check that , and if it exists you go on with your logic

Reply:
thanks for your feedback, appreciate it <3